ga4gh/datamodel/sequenceAnnotations.py

```diff
@@ -0,0 +1,415 @@
+        return feature
+
+    def getFeatures(self, referenceName, start, end,
+                    pageToken, pageSize,
+                    featureTypes=[], parentId=None):
+        """
+        Returns a set number of simulated features.
+        :param referenceName: name of reference to "search" on
+        :param start: start coordinate of query
+        :param end: end coordinate of query
+        :param pageToken: None or int
+        :param pageSize: None or int
+        :param ontologyTerms: optional list of ontology terms to limit query
+        :param parentId: optional parentId to limit query.
+        :return: Yields feature, nextPageToken pairs.
+            nextPageToken is None if last feature was yielded.
+
+        """
+        randomNumberGenerator = random.Random()
+        randomNumberGenerator.seed(self._randomSeed)
+        numFeatures = 10
+        if pageToken is not None:
+            nextPageToken = int(pageToken)
+        else:
+            nextPageToken = 0
+        for featureId in ["feature{}".format(x) for x in xrange(numFeatures)]:
+            gaFeature = self._generateSimulatedFeature(randomNumberGenerator)
+            gaFeature.id = self.getCompoundIdForFeatureId(featureId)
+            match = (gaFeature.start < end and
+                     gaFeature.end > start and
+                     gaFeature.referenceName == referenceName and
+                     (gaFeature.featureType in featureTypes or
+                      len(featureTypes) == 0))
+            if match:
+                gaFeature.parentId = ""  # TODO: Test with nonempty parentIDs?
+                if nextPageToken < numFeatures - 1:
+                    nextPageToken += 1
+                else:
+                    nextPageToken = None
+                yield gaFeature, \
+                    (str(nextPageToken) if nextPageToken is not None else None)
+
+
+class Gff3DbFeatureSet(AbstractFeatureSet):
+    """
+    Stub class to directly read sequence annotation features from GFF3 files.
+    Tests basic access, not to be used in production.
+    """
+    def __init__(self, parentContainer, localId, filePath, dataRepository):
+        super(Gff3DbFeatureSet, self).__init__(
+            parentContainer, localId, None)
+        self._sequenceOntology = dataRepository.getOntology(
+            'sequence_ontology')
+        self._dbFilePath = filePath
+        self._dataRepository = dataRepository
+        self._db = Gff3DbBackend(self._dbFilePath)
+
+    def getFeature(self, compoundId):
+        """
+        Returns a protocol.Feature object corresponding to a compoundId
+        :param compoundId: a datamodel.FeatureCompoundId object
+        :return: a Feature object.
+        :raises: exceptions.ObjectWithIdNotFoundException if invalid
+            compoundId is provided.
+        """
+        featureId = _recolonize(compoundId.featureId)
+        with self._db as dataSource:
+            featureReturned = dataSource.getFeatureById(featureId)
+
```

Comment:
I think it's better to seek out errors positively, and let the nominal case be the main route through the function. Here, e.g.,
```python
if featureReturned is None:
    raise exceptions.ObjectWithIdNotFoundException(compoundId)
return self._gaFeatureForFeatureDbRecord(featureReturned)
```
I would argue that this is a little bit clearer.